ReactiveCocoa/Swift/Property.swift

```diff
@@ -8,66 +8,486 @@ public protocol PropertyType {
 	/// The current value of the property.
 	var value: Value { get }
 
-	/// A producer for Signals that will send the property's current value,
-	/// followed by all changes over time.
+	/// A producer for signals that sends the property's current value,
+	/// followed by all changes over time. It completes when the property
+	/// has deinitialized, or has no further change.
 	var producer: SignalProducer<Value, NoError> { get }
 
-	/// A signal that will send the property's changes over time.
+	/// A signal that will send the property's changes over time. It
+	/// completes when the property has deinitialized, or has no further
+	/// change.
 	var signal: Signal<Value, NoError> { get }
+
+	/// Performs an arbitrary action using the current value of the
+	/// variable. Conforming types may optionally guarantee atomicity.
+	///
+	/// Returns the result of the action.
+	func withValue<Result>(@noescape action: Value throws -> Result) rethrows -> Result
```

Comment:
I don't think this is necessary. None of the property types guarantee atomicity with this PR.